lib/std/target.zig

```diff
@@ -1160,65 +1084,59 @@ pub const Target = union(enum) {
         }
     }
 
-    /// Caller owns returned memory.
-    pub fn getStandardDynamicLinkerPath(
-        self: Target,
-        allocator: *mem.Allocator,
-    ) error{
-        OutOfMemory,
-        UnknownDynamicLinkerPath,
-        TargetHasNoDynamicLinker,
-    }![:0]u8 {
-        const a = allocator;
+    /// The result will be a byte index *pointing at the final byte*. In other words, length minus one.
+    /// A return value of `null` means the concept of a dynamic linker is not meaningful for that target.
+    pub fn standardDynamicLinkerPath(self: Target, buffer: *[255]u8) ?u8 {
```

Comment:
Wouldn't it be more ziggy to return a slice?

Reply:
This actually ends up being more convenient to use in all the callsites, which are trying to avoid using an allocator. With this way there are no "invalid" states of the return value. Same for the struct fields which store a dynamic linker path, e.g. `NativeTargetInfo` and soon to be `CrossTarget` and `Target`.

Reply:
how is `?u8` better than`?[]u8` ?